### PR TITLE
test(food, user, auth, ping)

### PR DESCRIPTION
**Pull request by lauthieb**

### CODE REVIEW

**Review by landschootl — APPROVED** [review]

File: src/main/java/fr/foodlimit/api/security/MockSecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+
+@Profile("test")
+@Configuration
+public class MockSecurityConfig extends WebSecurityConfigurerAdapter {
+
+  private final TokenProvider tokenProvider;
```
Comment (lauthieb) [inline src/main/java/fr/foodlimit/api/security/MockSecurityConfig.java]: MAJOR Remove this unused "tokenProvider" private field. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/fr/foodlimit/api/security/MockSecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package fr.foodlimit.api.security;
+
+import fr.foodlimit.api.security.jwt.JWTConfigurer;
```
Comment (lauthieb) [inline src/main/java/fr/foodlimit/api/security/MockSecurityConfig.java]: MINOR Remove this unused import 'fr.foodlimit.api.security.jwt.JWTConfigurer'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/fr/foodlimit/api/security/MockSecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import fr.foodlimit.api.security.jwt.TokenProvider;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.HttpMethod;
```
Comment (lauthieb) [inline src/main/java/fr/foodlimit/api/security/MockSecurityConfig.java]: MINOR Remove this unused import 'org.springframework.http.HttpMethod'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by lauthieb** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**